packages/syft/src/syft/core/tensor/autodp/gamma_tensor.py

```diff
@@ -580,6 +580,71 @@ def sum(
 
         return result
 
+    def mean(
+        self,
+        *args: Tuple[Any, ...],
+        **kwargs: Any,
+    ) -> Union[TensorWrappedGammaTensorPointer, MPCTensor]:
+        """Apply the "mean" operation to "self"
+
+        Args:
+            y Tuple[Any, ...] : second operand.
+
+        Returns:
+            Union[TensorWrappedGammaTensorPointer,MPCTensor] : Result of the operation.
+        """
+        attr_path_and_name = "syft.core.tensor.tensor.Tensor.mean"
+        min_vals = self.min_vals.mean(*args, **kwargs)
+        max_vals = self.max_vals.mean(*args, **kwargs)
+
+        result = TensorWrappedGammaTensorPointer(
+            data_subjects=self.data_subjects,
```

Comment:
MAYBE: Could we call the mean operation on the datasubjects also?

Reply:
Great catch Rasswanth!